test/lib/buildFactory.test.js

```diff
@@ -245,10 +252,25 @@ describe('Build Factory', () => {
 
     describe('getInstance', () => {
         it('should encapsulate new, and act as a singleton', () => {
-            const f1 = BuildFactory.getInstance({ datastore });
-            const f2 = BuildFactory.getInstance({ datastore });
+            const f1 = BuildFactory.getInstance({ datastore, executor });
+            const f2 = BuildFactory.getInstance({ datastore, executor });
 
             assert.equal(f1, f2);
         });
+
+        it('should not require config on second call', () => {
+            const f1 = BuildFactory.getInstance({ datastore, executor });
+            const f2 = BuildFactory.getInstance();
+
+            assert.equal(f1, f2);
```

Comment:
This would verify the references of the object are identical, as opposed to the content, so it should be enough.